File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int neg = 0;
        for(auto num : nums){
            if(num < 0) neg++ ;
            if(num == 0) return 0;
        }
        return neg % 2 == 0 ? 1 : -1;
        
    }
};

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.rs
impl Solution {
    pub fn array_sign(nums: Vec<i32>) -> i32 {
        let mut neg : i32 = 0;
        for i in 0..nums.len() {
            if nums[i]==0 {
                return 0;
            }
            else if nums[i] < 0 {
                neg+=1;
            }
        }
        if neg%2 == 0 {
            return 1;
        }
        -1
    }
}